contracts/FeeBurner.sol

```diff
@@ -0,0 +1,271 @@
+
+import "./ERC20_token/ERC20.sol";
+import "./OMG_ERC20/OMG_ERC20.sol";
+import "./math/SafeMath.sol";
+
+/**
+ * @title FeeBurner
+ * @author Piotr Zelazko <pitor.zelazko@imapp.com>
+ * @dev TODO:
+ */
+contract FeeBurner { 
+    
+    using SafeMath for uint256;
+
+    /*
+    * Events
+    */
+    event ExchangeRateChanged(
+        address token,
+        uint blockNo,
+        uint nominator,
+        uint denominator
+    );
+
+    /*
+     * Storage
+     */
+
+    /*
+     * @notice the exchange rate is a ratio of OMG tokens to otherTokens 
+     * @notice ratio = OMGTokens/otherTokens = nominator/denominator
+     */
+    struct Rate {
+        uint nominator;
+        uint denominator;
+    } 
+
+    struct ExchangeRate {
+        uint blockNo;
+        Rate rate;
+    }
+
+    address public operator;
+    OMG_ERC20 public OMGToken;
+
+    uint constant public NEW_RATE_MATURITY_MARGIN = 5;
+
+    mapping (address => ExchangeRate) public previousExchangeRates;
+    mapping (address => ExchangeRate) public exchangeRates;
+
+    /*
+     * Modifiers
+     */    
+    modifier onlyOperator(){
+        require(msg.sender == operator);
+        _;
+    }
+
+    modifier checkRate(uint nominator, uint denominator){
+        require(nominator > 0);
+        require(denominator > 0);
+        _;
+    }
+
+    modifier supportedToken(address token){
+        require(exchangeRates[token].blockNo != 0);
+        _;
+    }
+
+    /**
+     * @dev Constructor
+     * 
+     * @param _OMGToken address of OMGToken contract
+     */
+    constructor(address _OMGToken)
+        public
+    {   
+        //TODO: should I check this ?
+        require(_OMGToken != address(0));
+
+        operator = msg.sender;
+        OMGToken = OMG_ERC20(_OMGToken);    
+
+    }
+
+    /*
+     * Public functions
+     */
+
+
+    /**
+     * @dev Receives Eth    
+     */
+
+    function receive() public payable {} //TODO: rename it to "()"?
+
+    /**
+     * @dev Adds support for some token
+     *
+     * @notice By setting _token address to 0, support for Ether can be added
+     *
+     * @param _token contract address of ERC20 token, or 0 for Ether, which support should be added
+     * @param _nominator nominator of initial exchange rate. See ExchangeRate struct.
+     * @param _denominator denominator of initial exchange rate. See ExchangeRate struct.
+     */
+    function addSupportFor(address _token, uint _nominator, uint _denominator)
+        public
+        onlyOperator
+        checkRate(_nominator, _denominator)
+    {
+            // Checks whether such a token hasn't been supported yet
+            require(exchangeRates[_token].blockNo == 0);
+
+            ExchangeRate memory exchangeRate = ExchangeRate(block.number, Rate(_nominator, _denominator));
+
+            previousExchangeRates[_token] = exchangeRate;
+            exchangeRates[_token] = exchangeRate;
+
+            //TODO: Should I emit an event ? Maybe sth like AddedSupport(token)
```

Comment:
yes and no. I think `ExchangeRateChanged` on both occasions is fine, the effect is the same, so apps tracking the exchange rates will have an easier job